server/http/controller.go

```diff
@@ -511,6 +531,117 @@ func (c *controller) ExecuteDefinition(ctx context.Context, testDefinition opena
 	return openapi.Response(http.StatusUnprocessableEntity, nil), nil
 }
 
+func (c *controller) executeTransaction(ctx context.Context, transaction tests.Transaction, runInfo openapi.RunInformation) (openapi.ImplResponse, error) {
+	// create or update transaction
+	transactionID := transaction.ID
+	resp, err := c.doCreateTransaction(ctx, transaction)
+	if err != nil {
+		if errors.Is(err, errTransactionExists) {
+			resp, err := c.doUpdateTransaction(ctx, transaction.ID, transaction)
+			if err != nil {
+				return resp, err
+			}
+		} else {
+			return resp, err
+		}
+	} else {
+		transactionID = id.ID(resp.Body.(openapi.Transaction).Id)
+	}
```

Comment:
We are not using the `transactionID` at the method start, so we could set it only after error handling
```suggestion
	// create or update transaction
	resp, err := c.doCreateTransaction(ctx, transaction)
	if err != nil {
		if errors.Is(err, errTransactionExists) {
			resp, err := c.doUpdateTransaction(ctx, transaction.ID, transaction)
			if err != nil {
				return resp, err
			}
		} else {
			return resp, err
		}
	} 
    
    transactionID := id.ID(resp.Body.(openapi.Transaction).Id)
```